install.ts

```diff
@@ -0,0 +1,234 @@
+  await mkdirp(DENO_BIN_DIR);
+  let args: string[];
+  if (isWindows) {
+    await rename(DENO_BIN, OLD_DENO_BIN);
+    args = [
+      "powershell.exe",
+      "-Command",
+      `Expand-Archive "${archive}" -DestinationPath "${DENO_BIN_DIR}"`
+    ];
+  } else {
+    args = ["gunzip", "-d", archive];
+  }
+  const child: Process = run({ args });
+  const childStatus: ProcessStatus = await child.status();
+  if (!childStatus.success) {
+    panic(Error(`(g)unzip failed. ${args} -> ${childStatus.code}`));
+  }
+  child.close();
+  if (!isWindows) {
+    if (platform.os === "linux") {
+      await rename(DENO_BIN, OLD_DENO_BIN);
+    }
+    await copyFile(archive.replace(/\.gz$/, ""), DENO_BIN);
+  }
+}
+
+async function makeHandy(): Promise<void> {
+  if (isWindows) {
+    if (!PATH.toLocaleLowerCase().includes(DENO_BIN_DIR.toLocaleLowerCase())) {
+      let updatedPath: string = PATH;
+      if (PATH.endsWith(";")) {
+        updatedPath += DENO_BIN_DIR;
+      } else {
+        updatedPath += `;${DENO_BIN_DIR}`;
+      }
+      const args: string[] = [
+        "powershell.exe",
+        "-Command",
+        `[Environment]::SetEnvironmentVariable("PATH","${updatedPath}",` +
+          `[EnvironmentVariableTarget]::User)`
+      ];
+      const ps: Process = run({ args });
+      const psStatus: ProcessStatus = await ps.status();
+      if (!psStatus.success) {
+        panic(Error(`Unable to edit PATH. ${args} -> ${psStatus.code}`));
+      }
+      ps.close();
+    }
+  } else {
+    await chmod(DENO_DIR, 0o744);
+    await chmod(DENO_BIN, 0o744);
+    if (!PATH.includes(DENO_BIN_DIR)) {
+      let bashProfile: string;
+      try {
+        bashProfile = new TextDecoder().decode(await readFile(BASH_PROFILE));
+      } catch (err) {
+        if (err.kind !== ErrorKind.NotFound) {
+          panic(err);
+        }
+        bashProfile = "";
+      }
+      if (!/PATH=[^\n]+.deno\/bin/.test(bashProfile)) {
+        bashProfile = `${bashProfile}\nPATH=$PATH:${DENO_BIN_DIR}\n`;
+        if (!/export PATH/.test(bashProfile)) {
+          bashProfile += "\nexport PATH\n";
+        }
+        await writeFile(BASH_PROFILE, new TextEncoder().encode(bashProfile));
+      }
+    }
+  }
+}
+
+async function checkVersion(tag: string): Promise<void> {
+  const denoProc: Process = run({
+    args: ["deno", "--version"],
+    stdout: "piped"
+  });
+  const denoStatus: ProcessStatus = await denoProc.status();
+  if (!denoStatus.success) {
+    panic(Error("Test run failed"));
+  }
+  const denoStdout: Uint8Array = new Uint8Array(32);
+  while ((await denoProc.stdout.read(denoStdout)).nread < 16);
+  denoProc.close();
+  const output: string = new TextDecoder().decode(denoStdout);
+  if (!RegExp(tag.replace(/^v/, "").replace(/\./g, "\\.")).test(output)) {
+    panic(Error("Version mismatch"));
+  }
+}
+
+async function main(): Promise<void> {
+  let tag: string;
+  if (/^v\d+\.\d+\.\d+$/.test(args[1])) {
```

Comment:
Please add a comment about what you're doing here. If args[1] exists but does not look like a version, you should print an error message and exit